jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/python/PythonLambdaPackager.kt

```diff
@@ -21,36 +24,47 @@ import java.util.concurrent.CompletionStage
 
 class PythonLambdaPackager : LambdaPackager {
     override fun createPackage(module: Module, file: PsiFile): CompletionStage<LambdaPackage> {
-        val future =
-            CompletableFuture<LambdaPackage>()
+        val future = CompletableFuture<LambdaPackage>()
         ApplicationManager.getApplication().executeOnPooledThread {
-            val virtualFile = file.virtualFile
-            val contentRoot = module.rootManager.contentRoots.find {
-                VfsUtilCore.isAncestor(
-                    it,
-                    virtualFile,
-                    true
-                )
-            }
-            if (contentRoot == null) {
-                future.completeExceptionally(RuntimeException("Unable to determine content root for $file"))
-                return@executeOnPooledThread
-            }
+            try {
+                val excludedRoots = mutableSetOf(*module.rootManager.excludeRoots)
 
-            val mappings = mutableMapOf<String, String>()
-            mappings[contentRoot.path] = "/"
+                // Keep the SDK out of the zip
+                val rootManager = ModuleRootManager.getInstance(module)
+                for (entry in rootManager.orderEntries) {
+                    excludedRoots.addAll(entry.getFiles(OrderRootType.CLASSES))
+                }
 
-            try {
-                val excludedRoots = module.rootManager.excludeRoots.toSet()
+                // Keep the venv out too
+                rootManager.sdk?.homeDirectory?.let { home ->
+                    PythonSdkType.getVirtualEnvRoot(home.path)?.let { root ->
+                        LocalFileSystem.getInstance().findFileByIoFile(root)?.let {
+                            excludedRoots.add(it)
+                        }
+                    }
+                }
+
+                val mappings = mutableMapOf<String, String>()
                 val packagedFile = createTemporaryZipFile { zip ->
-                    contentRoot.walkFiles(excludedRoots) { file ->
-                        file.inputStream.use { fileContents ->
-                            zip.putNextEntry(
-                                VfsUtilCore.getRelativeLocation(
-                                    file,
-                                    contentRoot
-                                )!!, fileContents
-                            )
+                    ModuleRootManager.getInstance(module).contentRoots.forEach { contentRoot ->
+                        contentRoot.walkFiles(excludedRoots) { file ->
+                            mappings[contentRoot.path] = "/"
+                            file.inputStream.use { fileContents ->
+                                zip.putNextEntry(VfsUtilCore.getRelativeLocation(file, contentRoot)!!, fileContents)
```

Comment:
I hate bang-bangs (`!!`) are we confident we're handling the null properly?

Reply:
Can't hurt to wrap it